torch/ao/quantization/fx/_lower_to_native_backend.py

```diff
@@ -371,77 +380,74 @@ def _lower_weighted_ref_functional(
-            if is_relu and q_relu_func is None:
-                continue
+    modules = dict(model.named_modules(remove_duplicate=False))
+    nodes = list(model.graph.nodes)
+    for n in model.graph.nodes:
 
-            # Set up match pattern: (dequantize - [relu_op - ] func_op - quantize)
-            # Func args: (dequantized inputs, dequantized weights[, bias])
-            # Quantize args: (func, scale, zp, dtype)
-            func_pattern: Tuple[Any, ...] = ()
-            if has_bias:
-                func_pattern = (ref_func, "dequantize", "dequantize", MatchAllNode)
-            else:
-                func_pattern = (ref_func, "dequantize", "dequantize")
-            if is_relu:
-                func_pattern = (F.relu, func_pattern)
-            pattern = (torch.quantize_per_tensor, func_pattern, MatchAllNode, MatchAllNode, MatchAllNode)
-
-            # Iterate through nodes in the graph to find a match
-            # If there is a match, replace the above pattern with the corresponding quantized op
-            modules = dict(model.named_modules(remove_duplicate=False))
-            nodes = list(model.graph.nodes)
-            for n in model.graph.nodes:
-                if not is_match(modules, n, pattern):
-                    continue
-                q_node = n
-                (func_node, output_scale_node, output_zp_node, dtype) = q_node.args
-                if is_relu:
-                    relu_node = func_node
-                    func_node = relu_node.args[0]
-                else:
-                    relu_node = None
-                input_dq_node = func_node.args[0]
-                weight_dq_node = func_node.args[1]
-
-                if should_skip_lowering(func_node, qconfig_map):
-                    continue
+        # Step 0: Find nodes that match this pattern (dequantize - functional op - quantize)
+        # We search for the pattern backwards, starting with the quantize node
+        # Quantize node args: (func, scale, zp, dtype)
+        if n.op != "call_function" or n.target != torch.quantize_per_tensor:
+            continue
+        q_node = n
+        (func_node, output_scale_node, output_zp_node, _) = q_node.args
+        # Handle cases where the functional op is wrapped in a ReLU
+        if func_node.target == F.relu:
+            relu_node = func_node
+            func_node = relu_node.args[0]
+        else:
+            relu_node = None
+        if should_skip_lowering(func_node, qconfig_map):
+            continue
+        # Linear args: (dequantized inputs, dequantized weights[, bias])
+        # Conv args: (dequantized inputs, dequantized weights[, bias, stride, padding, dilation, groups])
+        if func_node.op != "call_function" or func_node.target not in LOWER_FUNCTIONAL_MAP:
+            continue
+        (input_dq_node, weight_dq_node, *remaining_func_args) = func_node.args
+        if input_dq_node.target != "dequantize" or weight_dq_node.target != "dequantize":
```

Comment:
in general we should check for both op and target, op for dequantize is "call_method"